importer/helper.go

```diff
@@ -189,36 +186,26 @@ func loadGenesis(db idb.IndexerDb, in io.Reader) (err error) {
 }
 
 // InitialImport imports the genesis block if needed. Returns true if the initial import occurred.
-func InitialImport(db idb.IndexerDb, genesisJSONPath string, client *algod.Client, l *log.Logger) bool {
+func InitialImport(db idb.IndexerDb, genesisReader io.Reader, l *log.Logger) (bool, error) {
 	_, err := db.GetNextRoundToAccount()
-
 	// Exit immediately or crash if we don't see ErrorNotInitialized.
 	if err != idb.ErrorNotInitialized {
-		maybeFail(err, l, "getting import state, %v", err)
-		return false
+		if err != nil {
+			maybeFail(err, l, "getting import state, %v", err)
```

Comment:
maybeFail calls exit, so you don't need the if/else. Just call maybeFail first.

Reply:
Alternatively, you can change this to return an error instead of exiting as you did with checkGenesisHash. You can decide what you prefer here.

Reply:
I think returning an error is definitely better. Below, `else` is also not needed. The whole block can move left.